### PR TITLE
feat(profiling): add battery usage chart

### PR DESCRIPTION
**Pull request by JonasBa**
Adds a new battery usage chart to profiling. 

It seems like cocoa might report byte units for cpu_usage which seems odd? I'll verify and we can adjust it later

CleanShot 2023-08-29 at 19 30 06@2x

### CODE REVIEW

File: static/app/components/profiling/flamegraph/flamegraph.tsx [hunk 1]
```diff
@@ -209,6 +212,14 @@ function Flamegraph(): ReactElement {
     );
   }, [organization.features, profileGroup.metadata.platform]);
 
+  const hasBatteryChart = useMemo(() => {
+    const platform = profileGroup.metadata.platform;
+    return (
+      (platform === 'cocoa' || platform === 'android' || platform === 'node') &&
```
Comment (Zylphrex) [inline static/app/components/profiling/flamegraph/flamegraph.tsx:218]: why is node here? and does android report the battery as well?

File: static/app/components/profiling/flamegraph/flamegraph.tsx [hunk 2]
```diff
@@ -309,6 +320,26 @@ function Flamegraph(): ReactElement {
     hasUIFrames,
   ]);
 
+  const batteryChart = useMemo(() => {
+    if (!hasCPUChart) {
+      return LOADING_OR_FALLBACK_CPU_CHART;
+    }
```
Comment (Zylphrex) [inline static/app/components/profiling/flamegraph/flamegraph.tsx:326]: `hasBatteryChart` and `LOADING_OR_FALL_BACK_BATTERY_CHART?

**Review by Zylphrex — APPROVED** [review]
There's a few typos that you need to fix up but otherwise LGTM

File: static/app/components/profiling/flamegraph/flamegraph.tsx (hunk 1 above)
Comment (JonasBa) [reply static/app/components/profiling/flamegraph/flamegraph.tsx:218]: Heeem, no. This only needs to be cocoa

File: static/app/components/profiling/flamegraph/flamegraph.tsx (hunk 2 above)
Comment (JonasBa) [reply static/app/components/profiling/flamegraph/flamegraph.tsx:326]: Too fast copy pasta, nice find

**Pull request merged**